src/config/generate_function_list.sh

```diff
@@ -1,15 +1,16 @@
-#!/bin/sh
+#!/bin/bash
 # Copyright (c) 1998 Lawrence Livermore National Security, LLC and other
 # HYPRE Project Developers. See the top-level COPYRIGHT file for details.
 #
 # SPDX-License-Identifier: (Apache-2.0 OR MIT)
 
 # Print the defined function names in the object files of the current directory.
 #
-# The script uses 'nm' and searches for functions labeled with 'T'.
+# The script uses 'nm' and searches for functions labeled with 'T', i.e.,
+# symbol is in the text (code) section and is globally visible.
 
 # This prevents unmatched patterns from expanding (e.g., when there are no .obj files)
 shopt -s nullglob
 
-nm -P *.o *.obj | grep ' T ' | awk '{print $1}' | sed 's/^_//' | sed 's/_$//'
-
+# Use awk to avoid issues with spacing
+nm -P *.o *.obj | awk '$2 == "T" {print $1}' | sed -e 's/_$//'
```

Comment:
Why are you doing this?  The original scripts that Daniel wrote tried to use the position, but it wasn't portable because the position is not always the same ($2, $3, and $4 appeared to all be possible).  This was done intentionally.

Reply:
I just tried it on my mac, and I get failure (red UPDATE) for every directory.  I think it's the portability issue.

Reply:
Out of curiosity, did the script not work for you?  It works on my mac and my tux box, and it also works on Daniel's mac.  It's important to make this as portable as possible, but using position does not seem like a viable solution.

Reply:
Good point! This was in fact not portable. I just pushed a fix and verified it works on MacOS (Darwin) and linux